tests/main/interfaces-userns/task.yaml

```diff
@@ -0,0 +1,120 @@
+    sysctl -w kernel.unprivileged_userns_clone=1
+
+    echo "The userns interface is disconnected by default"
+    snap connections test-snapd-userns | MATCH 'userns +test-snapd-userns:userns +- +-'
+
+    # running unshare -U to create a new user namespace should be denied as an
+    # unprivileged user by seccomp
+    echo "When the userns interface is disconnected creating a new user namespace is denied"
+    tests.session -u test exec sh -c "snap run test-snapd-userns.with-userns-plug -c '\\$P/unshare -U id'" 2>&1 | MATCH 'unshare: unshare: Operation not permitted'
+
+    echo "Connect the userns interface"
+    snap connect test-snapd-userns:userns
+    snap connections test-snapd-userns | MATCH 'userns +test-snapd-userns:userns +:userns +manual'
+
+    # running unshare -U to create a new user namespace should be allowed
+    echo "When the userns interface is connected creating a new user namespace is allowed"
+    nobody_group=nogroup
+    # on arch the group is called nobody
+    if os.query is-arch-linux; then
+      nobody_group=nobody
+    fi
+    tests.session -u test exec sh -c "snap run test-snapd-userns.with-userns-plug -c '\\$P/unshare -U id'" 2>&1 | MATCH 'uid=65534\(nobody\) gid=65534\('"$nobody_group"'\) groups=65534\('"$nobody_group"'\)'
+
+    # check the userns feature is supported by snapd itself - this should always
+    # be the case when using the vendored apparmor
+    if snap debug sandbox-features --required apparmor:parser:snapd-internal; then
+      if ! snap debug sandbox-features --required apparmor:parser:userns; then
+        echo "The internal vendored apparmor within snapd does not support userns restrictions"
+        exit 1
+      fi
+    else
+      if ! snap debug sandbox-features --required apparmor:parser:userns; then
+        # apparmor in ubuntu since kinetic should support userns
+        if os.query is-ubuntu-ge 22.10; then
+          echo "Ubuntu host system AppArmor does not support userns but it should"
+          exit 1
+        fi
+        echo "Host system AppArmor does not support userns, skipping remainder of test"
+        exit 0
+      fi
+    fi
+
+    # now check the kernel AppArmor userns enforcement as well as seccomp
+    # enforcement above
+    if [ ! -e /proc/sys/kernel/apparmor_restrict_unprivileged_userns ]; then
+      echo "Skipping testing of AppArmor enforcement since it is not supported by the kernel"
+      exit 0
+    fi
+
+    # enable AppArmor enforcement
+    echo "Enable AppArmor enforcement of unprivileged user namespace restrictions"
+    sysctl -w kernel.apparmor_restrict_unprivileged_userns=1
+
+    # and remove the AppArmor permission from the snap's AppArmor profile
+    sed -i 's/userns,/##spread##userns,/' /var/lib/snapd/apparmor/profiles/snap.test-snapd-userns.with-userns-plug
+    tests.cleanup defer sed -i 's/^##spread##//' /var/lib/snapd/apparmor/profiles/snap.test-snapd-userns.with-userns-plug
```

Comment:
this shouldn't be needed as all in /var/lib/snapd is automatically restored during the generic cleanup (see tests/lib/state.sh)

Reply:
ah ok - also done in 18e5f8e2e9